aten/src/ATen/quantized/Quantizer.cpp

```diff
@@ -479,6 +480,43 @@ QTensorImpl* get_qtensorimpl(const Tensor& self) {
   return static_cast<QTensorImpl*>(self.unsafeGetTensorImpl());
 }
 
+// QNNPACK can access up to 8 bytes beyond the beginning of the tensor's storage
+// boundary which does trigger ASAN, and can result in a segfault if the memory
+// falls on an umapped page or one with no read permission.  Here we define a
+// custom allocator that allocates the extra storage required to keep this
+// behavior safe.
+struct QNNPACKAllocator final : at::Allocator {
```

Comment:
Are we making this the default allocator for all qtensors? Should we use this only when the backend engine is set to qnnpack and use the default allocator for fbgemm?

Reply:
This should be the default allocator for any input tensor that ends up in QNNPACK.

Reply:
All qtensors will take this path, so we should only use this when the backend is set to QNNPACK.

Reply:
Thanks for pointing that out Jerry.  This should be OK for FBGEMM as well.